### PR TITLE
Improve/ios8 popover

### PR DESCRIPTION
**Pull request by richardoti**
This PR is a way towards iPhone-compatibility -  #311
I've fully removed `UIPopoverController` which is deprecated since iOS 9 and
replaced it with `UIPopoverPresentationController ` which is available since iOS8.
Now it is possible to open popovers on iPhone.
The only drawback of   `UIPopoverPresentationController ` is that it's only supported on iOS8+, but it is not a problem, as we have iOS7 support branch.

This PR doesn't touch any aspects of design for iPhone, so the popovers look a bit weird if presented modally on iPhone, but I may think about particular implementation later.

Also, I think, it's good to provide additional Demo project with JSON specifically designed for iPhone. 
Now it looks like this:
ios simulator screen shot 27 aug 2015 13 58 59 ios simulator screen shot 27 aug 2015 13 59 03

On iPad nothing has changed:
screen shot 2015-08-27 at 14 06 13 screen shot 2015-08-27 at 14 06 06

### CODE REVIEW

**Comment by richardoti** [comment]
Aaaaah, OK. Tests succeeded, though.
screen shot 2015-08-27 at 15 10 25

File: Source/Cells/Popover/FORMPopoverFieldCell.m [hunk 1]
```diff
@@ -137,12 +126,23 @@ - (void)titleLabelPressed:(FORMFieldValueLabel *)titleLabel {
...
-                                                inView:self
-                              permittedArrowDirections:UIPopoverArrowDirectionUp | UIPopoverArrowDirectionDown
-                                              animated:YES];
+    self.contentViewController.modalPresentationStyle = UIModalPresentationPopover;
+    UIPopoverPresentationController *presentationController = [self.contentViewController popoverPresentationController];
+    presentationController.sourceView = self;
+    presentationController.sourceRect = self.bounds;
+
+    UIViewController *topViewController = [self topViewController];
+    [topViewController presentViewController:self.contentViewController animated:YES completion:nil];
```
Comment (3lvis) [inline Source/Cells/Popover/FORMPopoverFieldCell.m]: You need to do it in `topViewController`? `self` doesn't work?

Comment (richardoti) [inline Source/Cells/Popover/FORMPopoverFieldCell.m]: `self` is of type `FORMDateFieldCell `, so I need the topmost presented `UIViewController`

Comment (3lvis) [inline Source/Cells/Popover/FORMPopoverFieldCell.m]: Of course!

Comment (richardoti) [inline Source/Cells/Popover/FORMPopoverFieldCell.m]: So, why not to merge if it's OK?
Also, Im thinking of creating Demo for iPhone with form designed for it. I.e. the one we had listed in issues.

File: Source/Cells/Popover/FORMPopoverFieldCell.m [hunk 2]
```diff
@@ -137,12 +126,23 @@ - (void)titleLabelPressed:(FORMFieldValueLabel *)titleLabel {
...
+}
+
+- (UIViewController *) topViewController {
```
Comment (3lvis) [inline Source/Cells/Popover/FORMPopoverFieldCell.m]: Looks good, I just tried on the simulator :clap: 

Would you mind moving this method to a category on `UIViewController`?

Comment (richardoti) [inline Source/Cells/Popover/FORMPopoverFieldCell.m]: Done!

**Pull request merged**